src/plugins/data/public/ui/query_editor/query_editor.tsx

```diff
@@ -20,9 +20,14 @@ import { QueryEditorExtensions } from './query_editor_extensions';
 import { QueryEditorBtnCollapse } from './query_editor_btn_collapse';
 import { SimpleDataSet } from '../../../common';
 import { createDQLEditor, createDefaultEditor } from './editors';
+import { getQueryService } from '../../services';
+import { SIMPLE_DATA_SET_TYPES } from '../../../common';
 
-const LANGUAGE_ID = 'SQL';
-monaco.languages.register({ id: LANGUAGE_ID });
+const LANGUAGE_ID_SQL = 'SQL';
+monaco.languages.register({ id: LANGUAGE_ID_SQL });
+
+const LANGUAGE_ID_KUERY = 'kuery';
+monaco.languages.register({ id: LANGUAGE_ID_KUERY });
```

Comment:
nit: i dont think we should wait to register here. but nothing new for this PR.

fast follow or for 2.17 we refactor this all correctly and put it here:
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/5f19c37fadc98c05febc1010685280292fe58a0c/packages/osd-monaco/src/xjson/lexer_rules/index.ts#L38

Reply:
i'll try to refactor it for 2.17